modules/settings.js

```diff
@@ -13,63 +12,120 @@ function Settings() {
+    photos: 'boolean',
+    videos: 'boolean',
+    contacts: 'boolean',
+    text: 'boolean',
+    intervalTime: 'number',
+    id: 'number',
+    registeredTimer: 'boolean',
+    repeat: 'boolean'
+  };
 
-    if (myDictionary.videos !== undefined &&
-      typeof myDictionary.videos === 'boolean') {
-      options.videos = myDictionary.videos;
+  if (typeof potentialOptions === 'object') {
+    opts = potentialOptions; // validate parameter object
+  } else if (typeof potentialOptions === 'string') {
+    opts = {}; // single field check
+    if (typeof value === 'undefined') {
+      opts[potentialOptions] = this.options[potentialOptions];
+    } else {
+      opts[potentialOptions] = value;
     }
+  } else if (typeof potentialOptions === 'undefined') {
+    opts = this.options; // validate current options
+  } else {
+    // TODO - replace with ErrorHandler module
+    return console.log('Invalid validate parameter', field);
+  }
 
-    if (myDictionary.contacts !== undefined &&
-      typeof myDictionary.contacts === 'boolean') {
-      options.contacts = myDictionary.contacts;
-    }
+  // Support partial validation
+  for (var field in opts) {
+    if (typeof this.options[field] === 'undefined') {
+      // TODO - replace with ErrorHandler module
+      console.log('Unrecognized settings option', field);
+      valid = false;
+    } else if (typeof opts[field] !== validTypes[field]) {
+      // TODO - replace with ErrorHandler module
+      console.log('Invalid type for settings option', field);
 
-    if (myDictionary.text !== undefined &&
-      typeof myDictionary.text === 'boolean') {
-      options.text = myDictionary.text;
-    }
 
-    if (myDictionary.id !== undefined &&
-      typeof myDictionary.id === 'number') {
-      options.id = myDictionary.id;
-    }
+      // alert(field + ': ' + typeof opts[field] + ' vs ' + typeof validTypes[field]); //rmv
 
-    if (myDictionary.registeredTimer !== undefined &&
-      typeof myDictionary.registeredTimer === 'boolean') {
-      options.registeredTimer = myDictionary.registeredTimer;
-    }
 
-    if (myDictionary.repeat !== undefined &&
-      typeof myDictionary.repeat === 'boolean') {
-      options.repeat = myDictionary.repeat;
-    }
-    //////pass in the value in hours /////////
-    if (myDictionary.intervalTime !== undefined &&
-      typeof myDictionary.intervalTime === 'number') {
-      timeInMilliSec = myDictionary.intervalTime * 1000 * 60 * 60;
-      options.intervalTime = timeInMilliSec;
+      valid = false;
     }
+  }
 
-    localStorage.setItem('ffosbrOptions', JSON.stringify(options));
-    return options;
-  };
-}
+  return valid;
+};
+
+Settings.prototype.set = function(newOptions) {
+
+  if (typeof newOptions === 'undefined') {
+    return;
+  } else if (typeof newOptions !== 'object') {
+    // TODO - replace with ErrorHandler module
+    return console.log('Invalid config object', newOptions);
+  }
+
+  if (this.validate('photos', newOptions.photos)) {
```

Comment:
I think if we leave potentialOptions undefined, it'll validate all of our options at once, meaning we don't need this giant block of if statements. Using this would remove the ability to skip over invalid components of the options object while still setting the valid ones. Is this something we want? Or should we assume that if a field in this object is invalid, the object as a whole shouldn't be trusted?